src/connections.rs

```diff
@@ -19,19 +20,18 @@ use std::{net::SocketAddr, sync::Arc};
 use tokio::select;
 
 /// Connection instance to a node which can be used to send messages to it
+#[derive(Clone)]
 pub struct Connection {
     quic_conn: quinn::Connection,
```

Comment:
How about keeping this within the `ConnectionHandle` instance?, its name suggested me up front I'd be using this handle to make use of the connection and when dropped it internally handles the removal from the pool and disconnection accordingly.

Reply:
Hm, the purpose of `ConnectionHandle` is just to remove the connection from the pool. So maybe it should be called something else? Something like `ConnectionPoolRemover` or so?

> when dropped it internally handles the removal from the pool 

This would defy the purpose of connection pool, because then you would have to stash the connection handle somewhere, otherwise the connection would be closed. The idea is to now need any stash. Just fetch the connection by calling `Endpoint::connect_to` and it either returns the connection from the pool or opens a new one, but the caller doesn't need to worry about that.

Reply:
right, i am probably misunderstanding, I thought the pool has a clone of the quinn::Connection thus we could reach/use it through the ConnectionHandler, e.g. ConnectionHandler::get_conn() which gives a clone of quinn::Connection ...??....

Reply:
No, `ConnectionHandle` is only for removing. I'm going to rename it to make it more clear.